azurerm/resource_arm_cdn_endpoint.go

```diff
@@ -137,75 +191,131 @@ func resourceArmCdnEndpointCreate(d *schema.ResourceData, meta interface{}) erro
+			IsHTTPSAllowed:             &httpsAllowed,
+			IsCompressionEnabled:       &compressionEnabled,
+			QueryStringCachingBehavior: cdn.QueryStringCachingBehavior(cachingBehaviour),
+			OriginHostHeader:           utils.String(originHostHeader),
+		},
+		Tags: expandTags(tags),
 	}
 
-	origins, originsErr := expandAzureRmCdnEndpointOrigins(d)
-	if originsErr != nil {
-		return fmt.Errorf("Error Building list of CDN Endpoint Origins: %s", originsErr)
+	if optimizationType != "" {
+		endpoint.EndpointProperties.OptimizationType = cdn.OptimizationType(optimizationType)
 	}
-	if len(origins) > 0 {
-		properties.Origins = &origins
+	if originPath != "" {
+		endpoint.EndpointProperties.OriginPath = utils.String(originPath)
 	}
-
-	if v, ok := d.GetOk("origin_host_header"); ok {
-		host_header := v.(string)
-		properties.OriginHostHeader = &host_header
+	if probePath != "" {
+		endpoint.EndpointProperties.ProbePath = utils.String(probePath)
 	}
 
-	if v, ok := d.GetOk("origin_path"); ok {
-		origin_path := v.(string)
-		properties.OriginPath = &origin_path
+	origins, err := expandAzureRmCdnEndpointOrigins(d)
+	if err != nil {
+		return fmt.Errorf("Error Building list of CDN Endpoint Origins: %s", err)
 	}
-
-	if v, ok := d.GetOk("content_types_to_compress"); ok {
-		var content_types []string
-		ctypes := v.(*schema.Set).List()
-		for _, ct := range ctypes {
-			str := ct.(string)
-			content_types = append(content_types, str)
-		}
-
-		properties.ContentTypesToCompress = &content_types
+	if len(origins) > 0 {
+		endpoint.EndpointProperties.Origins = &origins
 	}
 
-	cdnEndpoint := cdn.Endpoint{
-		Location:           &location,
-		EndpointProperties: &properties,
-		Tags:               expandTags(tags),
+	future, err := client.Create(ctx, resourceGroup, profileName, name, endpoint)
+	if err != nil {
+		return fmt.Errorf("Error creating CDN Endpoint %q (Profile %q / Resource Group %q): %+v", name, profileName, resourceGroup, err)
 	}
 
-	future, err := client.Create(ctx, resGroup, profileName, name, cdnEndpoint)
+	err = future.WaitForCompletion(ctx, client.Client)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error waiting for CDN Endpoint %q (Profile %q / Resource Group %q) to finish creating: %+v", name, profileName, resourceGroup, err)
 	}
 
-	err = future.WaitForCompletion(ctx, client.Client)
+	read, err := client.Get(ctx, resourceGroup, profileName, name)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error retrieving CDN Endpoint %q (Profile %q / Resource Group %q): %+v", name, profileName, resourceGroup, err)
 	}
 
-	read, err := client.Get(ctx, resGroup, profileName, name)
+	d.SetId(*read.ID)
+
+	return resourceArmCdnEndpointRead(d, meta)
+}
+
+func resourceArmCdnEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	endpointsClient := meta.(*ArmClient).cdnEndpointsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
```

Comment:
This was changed to resourceGroup above.

Reply:
👍 fixed